### PR TITLE
Django 2.x compatibility

### PR DESCRIPTION
**Pull request by ofalk**
Import reverse from django.urls
Add on_delete to migration

(I have signed the contributor license agreement already)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ofalk! I've created [OSPR-2497](https://openedx.atlassian.net/browse/OSPR-2497) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/django-openid-auth/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by mduboseedx** [comment]
@edx/platform-team Could you give this a look?

File: django_openid_auth/views.py [hunk 1]
```diff
@@ -40,7 +40,10 @@
 from django.contrib.auth import (
     REDIRECT_FIELD_NAME, authenticate, login as auth_login)
 from django.contrib.auth.models import Group
-from django.core.urlresolvers import reverse
+try:
```
Comment (bmedx) [inline django_openid_auth/views.py]: Can you mark this as a shim with a comment, so we'll know when to remove it? Something like:

`# SHIM: For Django 1.11 and 2.x compatiblity, can remove when support for 1.11 is dropped.`

Comment (ofalk) [reply django_openid_auth/views.py]: @bmedx Thanks for the suggestion. shim added now!

**Comment by ofalk** [comment]
@bmedx In regards to testing. As per https://docs.djangoproject.com/en/2.0/faq/install/#what-python-version-can-i-use-with-django , Django 2.0 isn't supporting Python < 3.4, so testing with Python 2 doesn't make sense.
I'll try to get tox working and let you know!

**Comment by ofalk** [comment]
Hey all.
I've hacked around the differences in Python 3 (with Django 2.x and Django 1.x) vs. Python 2.x (Django 1.x). This isn't as trivial as it sounds, since the module does a lot of string handling, which leads to the situation where you have to deal with str, unicode or bytes...
I can imagine why nobody did this yet :smile: 
Anyway. I've created a separate branch on my fork and I'd love to get feedback from you guys:
https://github.com/ofalk/django-openid-auth/tree/django2-python3-django1-python2-compat

**Comment by bmedx** [comment]
@ofalk Thanks for doing that, and sorry for the delay in getting back to you. We are looking at the status of this fork and trying to determine a long term plan for supporting it, and whether we should try to reintegrate with the upstream trunk. We should have some more information on this and a merge plan soon.

**Comment by ofalk** [comment]
@bmedx Don‘t worry. It‘s sitting there for you guys to have a look. It should be fully compatible - at least from tox side.

**Comment by doctoryes** [comment]
@ofalk Thanks for your contribution! I've added a commit and rebased your mentioned branch on a branch I made here:

https://github.com/edx/django-openid-auth/tree/jeskew/oflak_django_20_compat

If you'd like to add my commit and submit a PR, then the PR can proceed.

**Comment by ofalk** [comment]
@doctoryes Merged your branch into mine. I hope it's OK this way now!

**Comment by doctoryes** [comment]
@ofalk great! Only one things remaining: If you could please squash your commits into a single commit with a descriptive message, I'd be happy to merge it.

**Comment by ofalk** [comment]
I'm not a huge fan of squashing, but I do you the favour :-)
Here you go, @doctoryes

**Comment by doctoryes** [comment]
Thanks, @ofalk . And thanks for your contribution! Merging now.

**Comment by edx-webhook** [comment]
@ofalk 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/django-openid-auth&entry.867055334=https://github.com/edx/django-openid-auth/pull/17&entry.1484655318=https://github.com/ofalk&entry.752974735=2018-07-02+08:19&entry.1917517419=2018-09-13+14:28&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by ofalk** [comment]
My pleasure! Thanks for merging!

**Pull request merged**